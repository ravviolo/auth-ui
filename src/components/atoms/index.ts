export { Button } from './Button/Button';
export { Checkbox } from './Checkbox/Checkbox';
export { Divider } from './Divider/Divider';
export { Heading } from './Heading/Heading';
export { IconButton } from './IconButton/IconButton';
export { InputField } from './InputField/InputField';
export { InputLabel } from './InputLabel/InputLabel';
export { TextButton } from './TextButton/TextButton';
