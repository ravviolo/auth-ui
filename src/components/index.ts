export * from './templates/LoginTemplate/LoginTemplate';
export * from './templates/SignUpTemplate/SignUpTemplate';
