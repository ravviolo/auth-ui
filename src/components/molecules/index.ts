export { AuthFooter } from './AuthFooter/AuthFooter';
export { AuthHeader } from './AuthHeader/AuthHeader';
export { AuthSocials } from './AuthSocials/AuthSocials';
export { EmailInput } from './EmailInput/EmailInput';
export { PasswordInput } from './PasswordInput/PasswordInput';
