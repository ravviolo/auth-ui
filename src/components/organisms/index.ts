export * from './LoginForm/LoginForm';
export * from './SignUpForm/SignUpForm';
