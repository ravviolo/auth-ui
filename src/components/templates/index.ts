export * from './LoginTemplate/LoginTemplate';
export * from './SignUpTemplate/SignUpTemplate';
