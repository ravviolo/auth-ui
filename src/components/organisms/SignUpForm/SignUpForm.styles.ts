import styled from 'styled-components';

import { Flex } from 'components/atoms';

export const Container = styled(Flex)`
  gap: 1em;
`;
