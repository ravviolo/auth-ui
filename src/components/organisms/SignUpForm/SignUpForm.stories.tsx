import { ComponentMeta, ComponentStory } from '@storybook/react';

import { SignUpForm } from 'components/organisms';

export default {
  title: 'Organisms/SignUpForm',
  component: SignUpForm,
} as ComponentMeta<typeof SignUpForm>;

export const Default: ComponentStory<typeof SignUpForm> = (args) => (
  <SignUpForm {...args} />
);
Default.storyName = 'SignUpForm';
