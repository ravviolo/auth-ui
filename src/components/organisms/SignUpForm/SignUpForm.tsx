import { useRef } from 'react';

import { Button } from 'components/atoms';
import { EmailInput, PasswordInput } from 'components/molecules';

import * as S from './SignUpForm.styles';

interface Props {
  onSubmitSignUpForm: (email: string, password: string) => void;
}

export const SignUpForm = ({ onSubmitSignUpForm }: Props) => {
  const emailRef = useRef<HTMLInputElement>(null);
  const passwordRef = useRef<HTMLInputElement>(null);

  const handleSubmit: React.FormEventHandler<HTMLFormElement> = (e) => {
    e.preventDefault();

    const email = emailRef.current?.value;
    const password = passwordRef.current?.value;

    if (email && password) {
      onSubmitSignUpForm(email, password);
    }
  };
  return (
    <form onSubmit={handleSubmit}>
      <S.Container direction="column">
        <EmailInput ref={emailRef} id="email" name="email" />
        <PasswordInput ref={passwordRef} id="password" name="password" />
        <Button label="Sign Up" type="submit" />
      </S.Container>
    </form>
  );
};
