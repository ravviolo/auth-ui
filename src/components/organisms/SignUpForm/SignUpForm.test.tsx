import { render, screen, within } from '@testing-library/react';
import userEvent from '@testing-library/user-event';
import { create } from 'react-test-renderer';

import { SignUpForm } from 'components/organisms';

import { Props as SignUpFormProps } from './SignUpForm';

describe('SignUpForm', () => {
  const mockOnSubmitSignUpForm = jest.fn<
    SignUpFormProps['onSubmitSignUpForm'],
    [string, string]
  >();
  const testEmail = 'test@test.com';
  const testPassword = 'testpassword';

  const props: SignUpFormProps = {
    testId: 'test-signup-form',
    onSubmitSignUpForm: mockOnSubmitSignUpForm,
  };

  it('should render email and password inputs', () => {
    render(<SignUpForm {...props} />);

    const emailInput = screen.getByTestId('email-input-signup-form');
    const passwordInput = screen.getByTestId('password-input-signup-form');

    expect(emailInput).toBeInTheDocument();
    expect(passwordInput).toBeInTheDocument();
  });

  it("should render submit button with 'Sign Up' text", () => {
    render(<SignUpForm {...props} />);

    const submitButton = screen.getByTestId('btn-submit-signup');

    expect(submitButton).toHaveAttribute('type', 'submit');
    expect(submitButton).toHaveTextContent('Sign Up');
  });

  it('should submit form with non-empty input fields - run submit handler once', async () => {
    render(<SignUpForm {...props} />);

    const submitButton = screen.getByTestId('btn-submit-signup');
    const emailInputField = within(
      screen.getByTestId('email-input-signup-form')
    ).getByTestId('input-field-email');
    const passwordInputField = within(
      screen.getByTestId('password-input-signup-form')
    ).getByTestId('input-field-password');

    await userEvent.type(emailInputField, testEmail);
    await userEvent.type(passwordInputField, testPassword);
    await userEvent.click(submitButton);

    expect(mockOnSubmitSignUpForm).toHaveBeenCalledTimes(1);
    expect(mockOnSubmitSignUpForm).toHaveBeenCalledWith(
      testEmail,
      testPassword
    );
  });

  it("shouldn't submit form with one or more empty input fields", async () => {
    render(<SignUpForm {...props} />);

    const submitButton = screen.getByTestId('btn-submit-signup');

    await userEvent.click(submitButton);

    expect(mockOnSubmitSignUpForm).not.toHaveBeenCalled();

    const emailInputField = within(
      screen.getByTestId('email-input-signup-form')
    ).getByTestId('input-field-email');

    await userEvent.type(emailInputField, testEmail);
    await userEvent.click(submitButton);

    expect(mockOnSubmitSignUpForm).not.toHaveBeenCalled();
  });

  it('should match snapshot', () => {
    const tree = create(<SignUpForm {...props} />).toJSON();

    expect(tree).toMatchSnapshot();
  });
});
