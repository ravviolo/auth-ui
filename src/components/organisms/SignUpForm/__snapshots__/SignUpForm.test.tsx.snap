// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SignUpForm matches snapshot 1`] = `
<form
  data-testid="test-signup-form"
  onSubmit={[Function]}
>
  <div
    className="sc-jSMfEi bQrDqy sc-idiyUo wwKtw"
    data-testid="flex-signup-form"
    direction="column"
  >
    <div
      className="sc-jSMfEi bQrDqy sc-breuTD jjaaRC"
      data-testid="email-input-signup-form"
      direction="column"
    >
      <label
        className="sc-papXJ daxRFt"
        data-testid="input-label-email"
        htmlFor="email"
      >
        Email
      </label>
      <input
        className="sc-ftvSup dLJLzu"
        data-testid="input-field-email"
        id="email"
        name="email"
        type="email"
      />
    </div>
    <div
      className="sc-jSMfEi bQrDqy sc-ksZaOG cRa-dNN"
      data-testid="password-input-signup-form"
      direction="column"
    >
      <label
        className="sc-papXJ daxRFt"
        data-testid="input-label-password"
        htmlFor="password"
      >
        Password
      </label>
      <input
        className="sc-ftvSup dLJLzu"
        data-testid="input-field-password"
        id="password"
        name="password"
        type="password"
      />
    </div>
    <button
      className="sc-bczRLJ jXOWFG"
      data-testid="btn-submit-signup"
      type="submit"
    >
      Sign Up
    </button>
  </div>
</form>
`;
